File: API/Entities/Product.cs
namespace API.Entities;

public class Product
{
    public Guid ProductId { get; set; }
    public string ProductName { get; set; }
    public string ProductDescription { get; set; }
    public int ProductPrice { get; set; }
    public string ProductType { get; set; }
    public string ProductBrand { get; set; }
    public int QuantityInStock { get; set; }
    public string PictureUrl { get; set; }
    public bool IsDeleted { get; set; }
}

File: Client/src/app/models/Product.ts
export type Product = {
    productId: string;
    productName: string;
    productDescription: string;
    productPrice: number;
    productType: string;
    productBrand: string;
    pictureUrl: string;
    quantityInStock: number;
    isDeleted: boolean;
};

export type ProductParam = {
    pageNumber: number;
    pageSize: number;
    sortBy: string;
    searchTerm: string;
    brandList: string[];
    typeList: string[];
};